src/Rings/PBWAlgebra.jl

```diff
@@ -20,7 +20,7 @@ mutable struct PBWAlgElem{T, S} <: NCRingElem
   sdata::Singular.spluralg{S}
 end
 
-mutable struct PBWAlgIdeal{D, T, S}
+mutable struct PBWAlgIdeal{D, T, S} <: Ideal{PBWAlgIdeal{T, S}}
```

Comment:
Should this be recursive? I expected something along the lines of `Ideal{PBWAlgebra{...}}`